src/components/SignButton/SignButton.tsx

```diff
@@ -74,26 +73,44 @@ export function SignButton() {
         }
 
         setSignStatus('SignError');
+      } finally {
+        isProcessing.off();
       }
-    } finally {
-      isProcessing.off();
-    }
-  }, [files, setFiles, setSignature, isProcessing]);
+    },
+    [files, setFiles, setSignature, isProcessing],
+  );
+
+  const handleNoWallet = useCallback(() => {
+    setSignStatus('No Sporran');
```

Comment:
Should we change this status to 'No Wallet' ?

Reply:
Yeah, let’s clean this up as well